### PR TITLE
Add an opt-in pre-commit hook to enforce flake8 being run.

### PR DESCRIPTION
**Pull request by okkays**
## What does this change?

(I've set this up locally, and figured it might be worth sharing)

- 🌎 We have CI/CD checks for flake8.
- ⛔ We want to make sure that we're not triggering CI/CD when we can easily check if it'll fail locally.
- ✅ This commit adds a git-hooks directory with a presubmit script that will run flake8, and check if it succeeds, before you can commit.
- ⚠️ Note that (thankfully) git won't run hooks without your permission, and folks who want to opt in to this will need to do a configuration step explained in git-hooks/README.md
- 🔮 Future commits might include a pre-push hook to enforce that the tests pass, as well.

I've tested this out locally, and it's something that's easily disabled/skipped as each developer **must opt into it** for it to take effect.

### Author

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [x] Check for, document, and establish a testing plan for any behavior that may vary across environments or is otherwise difficult to test.
+ [x] Check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [ ] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [ ] Check for any behavior that may vary across environments or is difficult to test, and ensure that it is well-understood, documented, and that there is a testing plan in place.
+ [ ] Re-check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

### CODE REVIEW

File: git-hooks/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+This directory contains hooks to avoid pushing broken or unpolished code to CI/CD, which can cause some noisy notifications.
+
+## Configuration
+
+To use them, you'll need to set your [gitconfig](https://git-scm.com/docs/githooks#_description) to include this directory for hooks. To do that for this repo only, run:
+
+```
+echo '[core]' >> .git/config;
+echo '  hooksPath="./git-hooks"' >> .git/config;
+```
```
Comment (billyhunt) [inline git-hooks/README.md:12]: Is this something that all the devs will set up on our own systems?

File: git-hooks/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+This directory contains hooks to avoid pushing broken or unpolished code to CI/CD, which can cause some noisy notifications.
+
+## Configuration
+
+To use them, you'll need to set your [gitconfig](https://git-scm.com/docs/githooks#_description) to include this directory for hooks. To do that for this repo only, run:
+
+```
+echo '[core]' >> .git/config;
+echo '  hooksPath="./git-hooks"' >> .git/config;
```
Comment (billyhunt) [inline git-hooks/README.md:11]: Since this requires a change to dev configuration, we should run this by @ilodidoj.

Perhaps discuss in the next pairing?

**Comment by okkays** [comment]
- Rename to .githooks

**Review by billyhunt — APPROVED** [review]
Looks good to me!

**Pull request merged**